lib/matplotlib/_pylab_helpers.py

```diff
@@ -65,7 +65,8 @@ def destroy(cls, num):
         if hasattr(manager, "_cidgcf"):
             manager.canvas.mpl_disconnect(manager._cidgcf)
         manager.destroy()
-        gc.collect(1)
+        del manager, num
+        gc.collect()
```

Comment:
The `del` is OK, but this should go back to being `gc.collect(1)`.  This was an intentional change in https://github.com/matplotlib/matplotlib/pull/3045 where we were running into performance issues where the user had a large number of objects and forcing frequently full cyclic collections was causing them problems. 

For the most part we should trust Python to do memory management and not try to "help"

```suggestion
        # Full cyclic garbage collection may be too expensive to do on every 
        # figure destruction https://github.com/matplotlib/matplotlib/pull/3045
        gc.collect(1)
```

Reply:
I firmly agree with not causing a performance regression, good catch. Actually, [this comment](https://github.com/matplotlib/matplotlib/pull/3045#issuecomment-42482288) in that pr corrected my understanding of the function, I thought it was only asking for gen 1 alone to be collected.